Ajo-Savings-Project/ajo-savings
# Chore/create seed data

**Description of PR**

- improve withPagination function - it now returns count and have default values for orderBy and order
- Revised the Group Table
- Revised Wallet and it's creation
- Created endpoints
    - /joinGroup 
    - /getGroup
    - /getGroups
- Created script to seed the db with users
    - Creates group 
    - Creates users

---

Please check all that apply.

#### Checklist:

- [x] Rebased with master
- [x] PR Description provided
- [ ] Did you communicate these changes with the affected teams?
- [x] Did unit tests/integration tests/E2E tests get properly added/updated?
- [x] Is this change production-ready including testing, feature toggles, configurations, and deployment considerations?

#### Frontend Checklist:

- [ ] Is this responsive and mobile friendly?
- [ ] Did you follow the [ARIA specs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA)

#### Backend Checklist:

- [x] Does the endpoint return the correct success/error format?

---

**Screenshots**  

- create group:
<img width="661" alt="image" src="https://github.com/Ajo-Savings-Project/ajo-savings/assets/48376769/a90e24f5-bfd7-4b3f-86bc-2e0d826d2370">

- getGroups
<img width="785" alt="image" src="https://github.com/Ajo-Savings-Project/ajo-savings/assets/48376769/1101ca0b-a674-4a53-9ea2-9630963b29de">

- joinGroup
<img width="543" alt="image" src="https://github.com/Ajo-Savings-Project/ajo-savings/assets/48376769/788c9bb4-76df-4e2c-8e94-89158b9a3e09">

- getGroup
<img width="711" alt="image" src="https://github.com/Ajo-Savings-Project/ajo-savings/assets/48376769/d956b1e9-8e7c-4347-a879-71243d72e696">

---

**How to test**  


1. To test DB Seeding
- navigate to the server directory in a new terminal (make sure you are in ../server)
- run yarn seed

**note: this will clear the db if run.**
